scripts/eosjs-port.js

```diff
@@ -0,0 +1,299 @@
+        data: {
+          account,
+          abi: serializedAbiHexString
+        }
+      }]
+    },
+    {
+      blocksBehind: 3,
+      expireSeconds: 30,
+    }, 3)
+
+    await sleep(100)
+    return res
+
+  }
+
+  async setcode ({ account, code, vmtype, vmversion }, { authorization }) {
+
+    const wasmHexString = code.toString('hex')
+    let [actor, permission] = authorization.split('@')
+
+    const res = await this.transaction({
+      actions: [{
+        account: 'eosio',
+        name: 'setcode',
+        authorization: [{
+          actor,
+          permission,
+        }],
+        data: {
+          account,
+          code: wasmHexString,
+          vmtype,
+          vmversion
+        }
+      }]
+    },
+    {
+      blocksBehind: 3,
+      expireSeconds: 30,
+    }, 3)
+
+    await sleep(100)
+    return res
+
+  }
+
+  async transaction (trx, trxConfig={}, numTries=0) {
+    trxConfig = { blocksBehind:3, expireSeconds:30, ...trxConfig }
+    let result
+    try {
+      result = await api.transact(trx, trxConfig)
+    } catch (err) {
+      const errStr = '' + err
+      if (errStr.toLowerCase().includes('exceeded by') && numTries > 0) {
+        console.error(errStr, ', retrying...')
+        await sleep(100)
+        return await this.transaction(trx, trxConfig, numTries-1)
+      }
+      throw err
+    }
+    return result
+  }
+
+  async getAccount (account) {
+    return rpc.get_account(account)
+  }
+
+  async getActions (account, pos, offset) {
+    return rpc.history_get_actions(account, pos, offset)
+  }
+
+  async updateauth ({ account, permission, parent, auth }, { authorization }) {
+
+    let [actor, perm] = authorization.split('@')
+
+    if ((parent == 'owner' && permission == 'owner') || parent == '') {
+      parent = '.'
+    }
+
+    auth.accounts.sort((a, b) => {
+      if (a.permission.actor <= b.permission.actor) {
+        return -1
+      }
+      return 1
+    })
+
+    const res = await this.transaction({
+      actions: [
+      {
+        account: 'eosio',
+        name: 'updateauth',
+        authorization: [{
+          actor,
+          permission: perm,
+        }],
+        data: {
+          account,
+          permission,
+          parent,
+          auth
+        },
+      }]
+    }, {
+      blocksBehind: 3,
+      expireSeconds: 30,
+    }, 3)
+
+    await sleep(100)
+    return res
+
+  }
+
+  async linkauth ({ account, code, type, requirement }, { authorization }) {
+
+    let [actor, permission] = authorization.split('@')
+
+    const res = await this.transaction({
+      actions: [{
+        account: 'eosio',
+        name: 'linkauth',
+        authorization: [{
+          actor,
+          permission,
+        }],
+        data: {
+          account,
+          code,
+          type,
+          requirement
+        },
+      }]
+    }, {
+      blocksBehind: 3,
+      expireSeconds: 30,
+    }, 3)
+
+    await sleep(100)
+    return res
+
+  }
+
+}
+
+module.exports = Eos
```

Comment:
Where does it come from?

Reply:
Hello, it comes from another pull request. It was necessary to update eosjs to the version 21, the version 16 was incompatible with the variant variables used in the document graph. As the version 16 is completely different from the version 21, the eosjs-port file exposes an interface similar to the eosjs v16, it only has the eosjs v16 methods used throughout the code.